app/templates/register.enter-your-details.html

```diff
@@ -5,21 +5,27 @@
 
 {% block main %}
 
-{% if errors.password or errors.password_confirm %}
-{% set errorType = errors.password[0] if errors.password else errors.password_confirm[0] %}
-<div class="panel panel--error">
-    <div class="panel__header">
-        <h1 class="panel__title venus">{{ errorType }}</h1>
-    </div>
-    <div class="panel__body" data-qa="error-body">
-        <p class="mars"><a class="js-inpagelink" href="#create-a-password">Please check the passwords and try again</a></p>
-    </div>
-</div>
-<br/>
-{% endif %}
+        {% if errors|length > 0 %}
+        <div class="panel panel--error">
+            <div class="panel__header">
+                {% if errors|length == 1 %}
+                <h1 class="panel__title venus">{{ errors[error][0] }}</h1>
+                {% elif errors|length > 1 %}
+                <h1 class="panel__title venus">There are {{ errors|length }} errors on this page</h1>
+                {% endif %}
+            </div>
+            <div class="panel__body" data-qa="error-body">
+                {% for error in errors %}
```

Comment:
I actually don't have any validation for password confirm since it'll all be done by the equalTo in the password field so we shouldn't ever get password_confirm here.